src/Appwrite/Platform/Workers/Functions.php

```diff
@@ -549,6 +549,7 @@ private function execute(
                     $headersFiltered[] = [ 'name' => $key, 'value' => $value ];
                 }
             }
+            $headersFiltered[] = ['name' => 'x-appwrite-execution-id', 'value' => $execution->getId()];
```

Comment:
I would add it to `$executionResponse['headers']` instead, so it doesnt have weird side-effects. We want it to behave like any other header